internal/index/manager/manager.go

```diff
@@ -1698,6 +1757,217 @@ func (mgr *Manager) triggerPcapProcessedWebhook(webhookUrl string, jsonBody []by
 	}
 }
 
+func writePcaps(pcapDir string, packets []pcapOverIPPacket) ([]string, error) {
+	filenames := []string(nil)
+	handledLinkTypes := map[layers.LinkType]struct{}{}
+	for len(packets) != 0 {
+		lt := packets[0].linkType
+		fnPartial := tools.MakeFilename("", "pcap")
+		fnFull := filepath.Join(pcapDir, fnPartial)
+		f, err := os.Create(fnFull)
+		if err != nil {
+			return filenames, err
+		}
+		defer func() {
+			if f != nil {
+				if err := f.Close(); err != nil {
+					log.Printf("error closing file %q: %v", fnFull, err)
+				}
+			}
+			if fnFull != "" {
+				log.Printf("removing file %q because of a previous error", fnFull)
+				if err := os.Remove(fnFull); err != nil {
+					log.Printf("error removing file %q: %v", fnFull, err)
+				}
+			}
+		}()
+		w, err := pcapgo.NewNgWriter(f, lt)
+		if err != nil {
+			return filenames, err
+		}
+		nextStart := 0
+		for i, packet := range packets {
+			if packet.linkType != lt {
+				if nextStart == 0 {
+					if _, ok := handledLinkTypes[lt]; !ok {
+						nextStart = i
+					}
+				}
+				continue
+			}
+			if err := w.WritePacket(packet.ci, packet.data); err != nil {
+				return filenames, err
+			}
+		}
+		if err := w.Flush(); err != nil {
+			return filenames, err
+		}
+		if f, err = nil, f.Close(); err != nil {
+			return filenames, err
+		}
+		filenames = append(filenames, fnPartial)
+		fnFull = ""
+		if nextStart == 0 {
+			break
+		}
+		packets = packets[nextStart:]
+		handledLinkTypes[lt] = struct{}{}
+	}
+	return filenames, nil
+}
+
+func (mgr *Manager) pcapOverIPPacketHandler() {
+	packets := []pcapOverIPPacket(nil)
+	queue := false
+	for {
+		select {
+		case packet := <-mgr.pcapOverIPPackets:
+			packets = append(packets, packet)
+			if queue {
+				continue
+			}
+			queue = true
+
+		case <-mgr.pcapOverIPFlush:
+			if len(packets) == 0 {
+				queue = false
+				continue
+			}
+		}
+		go func(packets []pcapOverIPPacket) {
+			filenames, err := writePcaps(mgr.PcapDir, packets)
+			if err != nil {
+				log.Printf("error writing PCAP-over-IP packets: %v", err)
+			}
+			if len(filenames) != 0 {
+				mgr.ImportPcaps(filenames)
+			}
+		}(packets)
+		packets = nil
+	}
+}
+
+func (mgr *Manager) newPcapOverIPEndpoint(ctx context.Context, address string) *pcapOverIPEndpoint {
```

Comment:
This is only called with `context.Background()` as parameter. Can't we ditch the parameter and get the context in here directly?

Reply:
yes we can, but it might be a good idea to switch more stuff to use a context, as that improves some stuff like clean shutdown, I'd leave it for now.